onmt/inputters/item.py

```diff
@@ -0,0 +1,280 @@
+    #  'tgt': {'tgt': ...},
+    #  'src_original': ['tok1', ...'tokn'],
+    #  'tgt_original': ['tok1', ...'tokm'],
+    #  'indices' : seq in bucket
+    #  'align': ...,
+    # }
+
+    return maybe_example
+
+
+def numericalize(vocabs, example):
+    """
+    """
+    numeric = example
+    numeric['src']['src_ids'] = []
+    if vocabs['data_task'] == ModelTask.SEQ2SEQ:
+        src_text = example['src']['src'].split()
+        numeric['src']['src_ids'] = vocabs['src'](src_text)
+        if example['tgt'] is not None:
+            numeric['tgt']['tgt_ids'] = []
+            tgt_text = example['tgt']['tgt'].split()
+            numeric['tgt']['tgt_ids'] = \
+                vocabs['tgt']([DefaultTokens.BOS] + tgt_text
+                              + [DefaultTokens.EOS])
+
+    elif vocabs['data_task'] == ModelTask.LANGUAGE_MODEL:
+        src_text = example['src']['src'].split()
+        numeric['src']['src_ids'] = \
+            vocabs['src']([DefaultTokens.BOS] + src_text)
+        if example['tgt'] is not None:
+            numeric['tgt']['tgt_ids'] = []
+            tgt_text = example['tgt']['tgt'].split()
+            numeric['tgt']['tgt_ids'] = \
+                vocabs['tgt'](tgt_text + [DefaultTokens.EOS])
+    else:
+        raise ValueError(
+                f"Something went wrong with task {vocabs['data_task']}"
+        )
+
+    if 'src_feats' in vocabs.keys():
+        for featname in vocabs['src_feats'].keys():
+            src_feat = example['src'][featname].split()
+            vf = vocabs['src_feats'][featname]
+            # we'll need to change this if we introduce tgt feat
+            numeric['src'][featname] = vf(src_feat)
+
+    return numeric
+
+
+def parse_align_idx(align_pharaoh):
+    """
+    Parse Pharaoh alignment into [[<src>, <tgt>], ...]
+    """
+    align_list = align_pharaoh.strip().split(' ')
+    flatten_align_idx = []
+    for align in align_list:
+        try:
+            src_idx, tgt_idx = align.split('-')
+        except ValueError:
+            logger.warning("{} in `{}`".format(align, align_pharaoh))
+            logger.warning("Bad alignement line exists. Please check file!")
+            raise
+        flatten_align_idx.append([int(src_idx), int(tgt_idx)])
+    return flatten_align_idx
+
+
+def tensorify(vocabs, minibatch):
+    """
+    This function transforms a batch of example in tensors
+    Each example looks like
+    {'src': {'src': ..., 'feat1': ..., 'feat2': ..., 'src_ids': ...},
+     'tgt': {'tgt': ..., 'tgt_ids': ...},
+     'src_original': ['tok1', ...'tokn'],
+     'tgt_original': ['tok1', ...'tokm'],
+     'indices' : seq in bucket
+     'align': ...,
+    }
+    Returns  Dict of batch Tensors
+        {'src': [seqlen, batchsize, n_feats],
+         'tgt' : [seqlen, batchsize, n_feats=1],
+         'indices' : [batchsize],
+         'srclen': [batchsize],
+         'tgtlen': [batchsize],
+         'align': alignment sparse tensor
+        }
+    """
+    tbatch = {}
```

Comment:
not fan of `tbatch`, not very readable and could stand for text batch as much as tensor batch